File: info-collector/plugins/twitter.py
#!/usr/bin/env python

from abstractplugin import AbsPlugin
import requests
import time


class TwitterPlugin(AbsPlugin):
    def __init__(self):
        pass

    def get_data(self,job):
        url = u'http://search.twitter.com/search.json?q='
        url = url + ' OR '.join(job.tags.split(' '))
        url = url + "&count=100"
        if job.lat > 0 and job.lon > 0:
            url = url + '&geocode=' + job.lat + ',' + job.lon + ',' + str(job.distance) + 'mi'
        twit = requests.get(url)
        data = []
        for tweet in twit.json()['results']:
            date =  time.strptime(tweet['created_at'], '%a, %d %b %Y %H:%M:%S +0000')
            data.append({
                'source': 'twitter',
                'id': str(tweet['id']),
                'creator': tweet['from_user'],
                'time': int(time.strftime('%s',date)),
                'location': tweet['location'] if 'location' in tweet else u'',
                'lat': tweet['geo']['coordinates'][0] if tweet['geo'] and 'coordinates' in tweet['geo'] else 0.0,
                'lon': tweet['geo']['coordinates'][1] if tweet['geo'] and 'coordinates' in tweet['geo'] else 0.0,
                'data': tweet['text']
            })

        return data


# Each plugin must provide a load method at module level that will be
# used to instantiate the plugin
def load():
    """ Loads the current plugin """
    return TwitterPlugin()

File: collector/src/main/scala/com/geocent/collector/sources/twitter.scala
package com.geocent.collector.sources

import com.geocent.collector._
import akka.actor.Actor
import dispatch._
import dispatch.liftjson.Js._


object Twitter extends InformationSource{
	def getData(job:Job):Seq[Information] = {
		val url = "http://search.twitter.com/search.json?q="
		+ job.tags.split(" ").mkString(" OR ")
		+ "&lang=en&count=100"
		+ job.lat.map { job.lat.get + "," + job.lon.get + "," + distance.get + "mi" }.getOrElse{""}

		val LARGE_TWITTER_DATE_FORMAT = "EEE MMM dd HH:mm:ss Z yyyy"
		val format = new java.text.SimpleDateFormat("dd-MM-yyyy")
		val str = Http(host(url) OK as.Json).option

		val results = for {
			JObject(result) <- str
			JField("id_str", JString(is_str)) <- child
			JField("created_at", JString(created_at)) <- child
			JField("from_user", JString(from_user)) <- child			
			JField("text", JString(text)) <- child
			JField("location", JString(location)) <- child
			JField("geo", JArray(geo)) <- child			
		} yield (id_str, created_at, from_user, text, location, geo)

		results.map{ result =>
			Information(
				source = "twitter",
				sourceId = result._1,
				creator = result._3,
				time = new SimpleDateFormat(LARGE_TWITTER_DATE_FORMAT, Locale.ENGLISH).parse(result_2).getMills
				)

		}

		return Seq[Information]()
	}
}

class TwitterActor extends Actor {
	override def preStart() = {
		println("Twitter Starting")
	}

	def receive = {
		case _	=> Twitter.execute
	}
}